src/main/java/com/faforever/api/leaderboard/GlobalLeaderboardRepository.java

```diff
@@ -23,14 +23,40 @@
     "     WHERE (expires_at is null or expires_at > NOW()) AND revoke_time IS NULL" +
     "  ) " +
     "  ORDER BY rating DESC LIMIT ?#{#pageable.offset},?#{#pageable.pageSize}",
-    countQuery = "SELECT count(*) FROM ladder1v1_rating WHERE is_active = 1 AND ladder1v1_rating.numGames > 0" +
+    countQuery = "SELECT count(*) FROM global_rating WHERE is_active = 1 AND global_rating.numGames > 0" +
       "   AND id NOT IN (" +
       "     SELECT player_id FROM ban" +
       "     WHERE (expires_at is null or expires_at > NOW()) AND revoke_time IS NULL" +
       "       AND (1=1 OR -1 IN (?,?,?))" +
       "  ) -- Dummy placeholder for pageable, prevents 'Unknown parameter position': ?,?,?", nativeQuery = true)
   Page<GlobalLeaderboardEntry> getLeaderboardByPage(Pageable pageable);
 
+  @Query(value = "SELECT" +
+    "    global_rating.id," +
+    "    login.login," +
+    "    global_rating.mean," +
+    "    global_rating.deviation," +
+    "    global_rating.numGames," +
+    "    @s \\:= @s + 1 rank" +
+    "  FROM global_rating JOIN login on login.id = global_rating.id," +
+    "    (SELECT @s \\:= ?#{#pageable.offset}) AS s" +
+    "  WHERE is_active = 1 AND global_rating.numGames > 0" +
+    "   AND login.id NOT IN (" +
+    "     SELECT player_id FROM ban" +
+    "     WHERE (expires_at is null or expires_at > NOW()) AND revoke_time IS NULL" +
+    "  ) " +
+    "   AND login.login LIKE ?#{#playerNameRegex}"+
```

Comment:
interesting that this works. but I would prefer `:playerNameRegex`

Reply:
that does not work. U are supposed to use one ways to reference parameters only. I go for Spel type of reference. Your suggestion does not work... I tried

Reply:
thanks good to know